obspy/core/stream.py

```diff
@@ -795,13 +795,13 @@ def getGaps(self, min_gap=None, max_gap=None):
                 continue
             # different sampling rates should always result in a gap or overlap
             if self.traces[_i].stats.delta == self.traces[_i + 1].stats.delta:
-                flag = True
+                same_sampling_rate = True
             else:
-                flag = False
+                same_sampling_rate = False
             stats = self.traces[_i].stats
             stime = stats['endtime']
             etime = self.traces[_i + 1].stats['starttime']
-            delta = etime.timestamp - stime.timestamp
+            delta = etime.timestamp - (stime.timestamp + stats.delta)
```

Comment:
While correct this is a bit confusing to me. This would be more clear to me but maybe we just have different processes of thought ;-)

```python
delta = (etime - stime) - stats.delta
```